packages/salesforce-adapter/src/transformers/xml_transformer.ts

```diff
@@ -182,36 +323,52 @@ export const fromRetrieveResult = async (
   const fromZip = async (
     zip: JSZip, file: FileProperties,
   ): Promise<MetadataValues | undefined> => {
-    const getFileData = (name: string): Promise<Buffer | undefined> => {
-      const zipFile = zip.file(`${PACKAGE}/${name}`)
-      return zipFile === null
-        ? Promise.resolve(undefined)
-        : zipFile.async('nodebuffer')
+    const extractFileNameToData = async (fileName: string, withMetadataSuffix: boolean,
+      complexType: boolean): Promise<Record<string, Buffer>> => {
+      if (!complexType) { // this is a single file
+        const zipFile = zip.file(`${PACKAGE}/${fileName}${withMetadataSuffix ? METADATA_XML_SUFFIX : ''}`)
+        return zipFile === null ? {} : { [zipFile.name]: await zipFile.async('nodebuffer') }
+      }
+      // bring all matching files from the fileName directory
+      const zipFiles = zip.file(new RegExp(`^${PACKAGE}/${fileName}/.*`))
+        .filter(zipFile => zipFile.name.endsWith(METADATA_XML_SUFFIX) === withMetadataSuffix)
+      return _.isEmpty(zipFiles)
+        ? {}
+        : _.fromPairs(await Promise.all(zipFiles.map(async zipFile => [zipFile.name, await zipFile.async('nodebuffer')])))
```

Comment:
relevant in a lot of places:
prefer built in language features over using lodash
```suggestion
        : Object.fromEntries(await Promise.all(zipFiles.map(async zipFile => [zipFile.name, await zipFile.async('nodebuffer')])))
```